app/src/main/java/com/example/weatherapp/ui/MainActivity.kt

```diff
@@ -31,6 +25,8 @@ class MainActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        activityViewModel.deleteUnsavedData()
```

Comment:
Make the delete call in the `init` of the viewModel. It is unnecessary for the activity to know about this therefore it shouldn't know.